src/controller/expenses.controller.js

```diff
@@ -0,0 +1,107 @@
+const expensesService = require('../services/expenses.service');
+const userService = require('../services/user.service');
+
+const get = (req, res) => {
+  const { userId, from, to, categories } = req.query;
+
+  if (!userId && !categories && (!from || !to)) {
+    res.send(expensesService.getAllExpenses());
+  }
+
+  const normalizedCategories = expensesService.normalizeCategory(categories);
+
+  let filteredExpenses = expensesService.getAllExpenses();
+
+  if (userId) {
+    filteredExpenses = expensesService.filterExpensesById(userId);
+  }
+
+  if (normalizedCategories) {
+    filteredExpenses =
+      expensesService.filterExpensesByCategory(normalizedCategories);
+  }
+
+  if (from && to) {
+    const fromDate = new Date(from);
+    const toDate = new Date(to);
+
+    filteredExpenses = expensesService.filterExpensesByDate(fromDate, toDate);
+  }
```

Comment:
Here again, you are overwriting the `filteredExpenses` variable. Consider using the `filter` method of the array to only select the expenses within the given date range.